tukaan/_props.py

```diff
@@ -19,182 +26,145 @@ def config(widget: TkWidget, **kwargs) -> None:
+        ...
 
 
-def get_text(self) -> str:
-    return cget(self, str, "-text")
+class WidgetDesc(RWProperty[T, T_contra]):
+    def __init__(self, command: str, type: type[T]):
+        self._command = command
+        self._type = type
 
+    def __get__(self, instance: TkWidget, owner: object = None) -> T:
+        if owner is None:
+            return NotImplemented
+        return cget(instance, self._type, f"-{self._command}")
 
-def set_text(self, value: str) -> None:
-    config(self, text=value)
+    def __set__(self, instance: TkWidget, value: T_contra):
+        config(instance, **{self._command: value})
 
 
-text = property(get_text, set_text)
+class BoolDesc(WidgetDesc[bool, bool]):
+    def __init__(self, command: str):
+        super().__init__(command, bool)
 
 
-def get_width(self) -> int:
-    return cget(self, int, "-width")
+class FloatDesc(WidgetDesc[float, float]):
+    def __init__(self, command: str):
+        super().__init__(command, float)
 
 
-def set_width(self, value: int) -> None:
-    config(self, width=value)
+class Compound(WidgetDesc[ImagePosition, ImagePosition]):
+    def __init__(self):
+        super().__init__("compound", ImagePosition)
 
 
-width = property(get_width, set_width)
+class TextAlign(WidgetDesc[Justify, Justify]):
+    def __init__(self):
+        super().__init__("justify", Justify)
 
 
-def get_height(self) -> int:
-    return cget(self, int, "-height")
+class Foreground(WidgetDesc[Color, Color | str]):
+    def __init__(self):
+        super().__init__("foreground", Color)
 
 
-def set_height(self, value: int) -> None:
-    config(self, height=value)
+class Background(WidgetDesc[Color, Color | str]):
+    def __init__(self):
+        super().__init__("background", Color)
 
 
-height = property(get_height, set_height)
+class Text(WidgetDesc[str, str]):
+    def __init__(self):
+        super().__init__("text", str)
 
 
-def get_command(self) -> Callable | None:
-    return _commands.get(cget(self, str, "-command"))
+class Width(WidgetDesc[int, int]):
+    def __init__(self):
+        super().__init__("width", int)
 
 
-def set_command(self, func: Callable | None) -> None:
-    value = func
-    if value is None:
-        value = ""
-    config(self, command=value)
+class Height(WidgetDesc[int, int]):
+    def __init__(self):
+        super().__init__("height", int)
 
 
-command = property(get_command, set_command)
+class Command(RWProperty[Callable | None, Callable | None]):
+    def __get__(self, instance: TkWidget, owner: object = None):
+        if owner is None:
+            return NotImplemented
+        return _commands.get(cget(instance, str, "-command"))
 
+    def __set__(self, instance: TkWidget, value: Callable | None = None):
+        config(instance, command=value or "")
 
-def get_orientation(self) -> Orientation:
-    return cget(self, Orientation, "-orient")
 
+class Orient(WidgetDesc[Orientation, Orientation]):
```

Comment:
`class Orientation`?

Reply:
Orientation is already taken up by the enum. I think it should either be OrientProp or OrientDesc

Reply:
Let's go with `OrientProp`